src/staticdata.c

```diff
@@ -1799,6 +1799,44 @@ static void jl_set_nroots_sysimg(void)
 static void jl_init_serializer2(int);
 static void jl_cleanup_serializer2(void);
 
+// NOTE: currently protected by the codegen lock, since it is only used there
+jl_array_t *jl_global_roots_table;
```

Comment:
cc: @pchintalapudi

Reply:
Should I give this its own lock instead? It would be a leaf lock.

Reply:
Easy enough, so I just went ahead and did that.